### PR TITLE
Deletes all gamemodes in favor of storytellers

### PR DESCRIPTION
**Pull request by Moltijoe**
This is an extension of https://github.com/yogstation13/Yogstation/pull/22565
That pr works (mostly) and is stable enough to leave testmerged for the time being

This pr goes further, removing all gamemodes as storytellers make them obsolete.
Removing gamemodes will improve code maintainability in the future, as well as reduce code bloat.
However, a number of unrelated features relied on game modes to work, so while removing them, i noticed that it would become far more unstable for the time being.

Rather than risk messing up that original pr, i made a new branch, that is this

Now for how the system itself works

# Storytellers
It works by using the existing event system that was previously only for midround events, but develops it further
It has 5 different event tracks
- Mundane - Barely affects the round, whether positive or negative
- Moderate - Can have an affect on the round, but usually isn't outright destructive
- Major - Can have significant lasting effects on the round, some lesser ghost role antags are under this track
- Roleset - Contains the majority of antags, whether ghost role or otherwise
- Objectives - An objective for the crew to complete that rewards them in some way

Each event also has a list of tags, such as 
- TAG_ALIEN - The event is an alien threat
- TAG_COMMUNAL - The event impacts most of the crewmembers
- TAG_DESTRUCTIVE - the event generally causes destruction to the station
- TAG_SPOOKY - the event can be considered somewhat scary in thematic
- Others - it's easy to add more tags, allowing for more event selection customization

The storytellers work by having a "storyteller" generate points over time for each track
Once a track reaches a certain threshold, the "storyteller" will buy an event from that track
The event purchased will be determined by
- The original designated weight
- How many times it's been run before during the round
- How the storyteller favours that event's tags

Different types of storytellers can be made to convey different game feelings
- Guide - regular experience
- Sleeper - less antags, but more events that are less likely to be destructive or combatitive
- Mystic - more spooky, magical, or space related events
- Operative - More antags, and those antags are more likely to come from within the crew, can spawn roundstart antags during the round
- Clown - admin only - Spawns WAY more mundane and moderate events and they can repeat freely, but far fewer major events

Adding new events to this system is easier than making new event for dynamic, and they're going to spawn in more varied amounts compared to the old events + gamemode system

# Why is this good for the game?
Dynamic is less dynamic than would be ideal and requires more work to update, maintain, and improve that we have
Storytellers is both more flexible to start, and is a lot easier to understand from a coding perspective, making it far less of a resource drain on our dev team to maintain

# Testing
why should i need to test this

:cl: Biome, Molti, Dwasint, & other Monke devs
rscadd: Adds storytellers
rscdel: Deletes literally all gamemodes, buh bye
tweak: so many round events, like, genuinely so much
/:cl:

### CODE REVIEW

**Comment by teh4565** [comment]
Gone, reduced to atoms

**Comment by UselessTheremin** [comment]
you have achieved rp. yogstation has been saved. YES

File: code/modules/events/tzimisce.dm [hunk 1]
```diff
@@ -4,42 +4,24 @@
 	max_occurrences = 2
 	min_players = 25
 	earliest_start = 45 MINUTES
+	track = EVENT_TRACK_ROLESET
+	tags = list(TAG_SPOOKY, TAG_MAGICAL, TAG_COMBAT)
+	category = EVENT_CATEGORY_ENTITIES
+	description = "spawns a bloodsucker from ghost roles."
```
Comment (tattax) [inline code/modules/events/tzimisce.dm]: it is a specific clan

**Review by Bibby0110 — DISMISSED** [review]
can we get an actual description pls :3

File: code/modules/events/tzimisce.dm (hunk 1 above)
Comment (Moltijoe) [reply code/modules/events/tzimisce.dm]: fixed

**Comment by Moltijoe** [comment]
> can we get an actual description pls :3

certainly

**Comment by ktlwjec0** [comment]
grav anomaly announces itself as a flux anomaly

**Review by ToasterBiome — APPROVED** [review]
Ready

**Pull request merged**